src/Common/ZooKeeper/examples/zkutil_test_async.cpp

```diff
@@ -17,6 +17,8 @@ try
 
     size_t num_threads = DB::parse<size_t>(argv[1]);
     std::vector<std::thread> threads;
+    threads.reserve(num_threads);
```

Comment:
Does `.reserve` fix a clang-tidy warning? The thing is that the internal reserve (memory allocation) happens when the vector is created (l. 19) and at the latest in the first insert (l. 24). In all reasonable standard libraries, more space than the minimally needed space is reserved, so future inserts don't allocate again. Assuming that `num_threads` is "small", we will certainly allocate only once. So an explicit `.reserve` seems not really necessary.

Reply:
>Does `.reserve` fix a clang-tidy warning?

Yes, it does.